### PR TITLE
chore: add home-manager installation.

### PR DESCRIPTION
**Pull request by soasme**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,6 +159,120 @@ require('avante').setup ({
 
 </details>
 
+<details>
+
+  <summary>home-manager</summary>
+
+```nix
```
Comment (aarnphm) [inline README.md:166]: I would prefer just implement a flake.nix, rather than this.

**Review by aarnphm — CHANGES_REQUESTED** [review]
lets do flake.nix instead

**Comment by soasme** [comment]
@aarnphm we could update nixpkgs github repo and add this package so that fromGitHub call is removed.

**Comment by aarnphm** [comment]
honestly I don't know even having nixpkgs is a good idea yet, we are iterating a lot faster than nix release cycle (ofc you can use unstable branch on nix, and right now it just seems like there will be too much debt we will have to maintain).

ofc you are free to use this on your own. Just that idk if it makes sense rn to add this in.

**Comment by aarnphm** [comment]
I think this PR can be a great reference for those who wants to use it. I will just close this for now.

**Comment by mrjbj** [comment]
I have a nix home manager (standalone) setup and tried for 2 days to get avante.nvim working in my config.  Closest I could get was to create a shell.nix file that setup a dev environment I could use to build the repo locally and then manually move it into nvim.  I works, but it's complicated and I would _love_ it if someone knew how to get it working as an input flake or homeManagerModule.   

I took my nix-shell environment and tried to create a flake out of it, but it kept choking on the mlua dependency for some reason.   Here's the code I used to try and make it work (see below), perhaps someone knows more about nix than me and can get it working.  I know there's a lot of updates forthcoming to the code but it would be great to be able to just `nix flake update` and have the latest version of avante.nvim up and running.  Thanks for any help as this is a really good way to stay on top of all of your progress.   Either way, I'm loath to move to Cursor AI so really appreciate your work here.  Thanks!

```
{
  description = "Flake to build yetone/avante.nvim via 'make BUILD_FROM_SOURCE=true'";
  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
    rust-overlay.url = "github:oxalica/rust-overlay";
    flake-utils.url = "github:numtide/flake-utils";
    mlua = {
      url = "github:mlua-rs/mlua";
      flake = false;
    };
    avante-nvim = {
      url = "github:yetone/avante.nvim/main";
      inputs.mlua.follows = "mula";
      flake = false;
    };
  };
  outputs = {
    nixpkgs,
    rust-overlay,
    flake-utils,
    avante-nvim,
    ...
  }:
    flake-utils.lib.eachDefaultSystem (system: let
      overlays = [(import rust-overlay)];
      pkgs = import nixpkgs {
        inherit system overlays;
      };
      rust = pkgs.rust-bin.stable."1.80.0".default;
      buildInputs = with pkgs; [
        rust
        openssl
        pkg-config
        cargo-edit
        cargo-watch
        rustfmt
        clippy
      ];
      buildEnv = pkgs.mkShell {
        nativeBuildInputs = buildInputs;
        shellHook = ''
          export OPENSSL_DIR="${pkgs.openssl.dev}"
          export OPENSSL_LIB_DIR="${pkgs.openssl.out}/lib"
          export OPENSSL_INCLUDE_DIR="${pkgs.openssl.dev}/include"
          export RUSTFLAGS="-C target-cpu=native"
        '';
      };
      avante-nvim-plugin = pkgs.vimUtils.buildVimPlugin {
        pname = "avante-nvim";
        version = "0.1.0";
        src = avante-nvim;
        buildPhase = ''
          echo "Starting build phase..."
          export PATH="${pkgs.lib.makeBinPath buildInputs}:$PATH"
          ${buildEnv.shellHook}
          make BUILD_FROM_SOURCE=true
          echo "Build phase completed."
        '';
        installPhase = ''
          echo "Starting install phase..."
          mkdir -p $out/share/vim-plugins/avante-nvim
          cp -r . $out/share/vim-plugins/avante-nvim
          echo "Copied all files to $out/share/vim-plugins/avante-nvim"
          echo "Install phase completed. Contents of $out/share/vim-plugins/avante-nvim:"
          ls -R $out/share/vim-plugins/avante-nvim
        '';
      };
    in {
      packages.default = avante-nvim-plugin;
      # ... rest of the outputs remain the same ...
    });
}
```

**Comment by soasme** [comment]
For whomever interested, this is the my minimal nix setup: https://github.com/soasme/m2-nix-config/commit/66371055b9986fcb86a6fc1764de3167925ff339
Hope it's helpful too.

**Pull request closed without merging**